### ISSUE TITLE
OWASP Scan Intermittently Fails

### ISSUE THREAD

**Issue by carltonsmith** [raft-tech/TANF-app#737]
**Description:**
There appears to be a problem with the OWASP check. The problem, though, is with the check itself and it invariably passes by just re-running the CI/CD process. We need to see why the check intermittently chokes

**Acceptance Criteria:** 

- [ ] OWASP check stops choking (this will be hard to determine immediately as it only intermittently fails. Let's discuss how evaluate this AC)

**Tasks:**

- [ ] Investigate root cause
- [ ] Determine course of action
- [ ] Implement fix
- [ ] Run Testing Checklist and confirm all tests pass

**Notes:**
- The cause is not known at this time, so we will update any changes in the estimate based on what we discover here

**Supporting Documentation:**

-  https://app.circleci.com/pipelines/github/raft-tech/TANF-app/1999/workflows/79b812e1-7f76-4434-a65e-85551ed6a1b2/jobs/4103

**Open Questions:**

- How can we effectively evaluate the AC, or rephrase the AC into something that can be effectively evaluated

**Comment by jtwillis92** [comment]
This occurred again but with a different intermittent failure than the Selenium issue noted before. This time it appears to be a connection error with the container that runs the zap scanner:
```
2021-04-16 18:23:11,529 I/O error: [Errno 5] Failed to connect to ZAP after 600 seconds
Traceback (most recent call last):
  File "/zap/zap-full-scan.py", line 326, in main
    wait_for_zap_start(zap, timeout * 60)
  File "/zap/zap_common.py", line 309, in wait_for_zap_start
    raise IOError(
OSError: [Errno 5] Failed to connect to ZAP after 600 seconds
```

**Comment by jtwillis92** [comment]
I did notice that we are doing a [ZAP Full Scan](https://www.zaproxy.org/docs/docker/full-scan/) which according to the docs

> It runs the ZAP spider against the specified target (by default with no time limit) followed by an optional ajax spider scan and then a full active scan before reporting the results.

> This means that the script does perform actual ‘attacks’ and can potentially run for a long period of time.

In referencing other government projects using this tool I noticed that they all tend to utilize [ZAP Baseline Scan](https://www.zaproxy.org/docs/docker/baseline-scan/) during CI/CD:

* https://github.com/HHS/Head-Start-TTADP/blob/main/bin/run-owasp-scan#L27
* https://github.com/usdoj-crt/crt-portal/blob/develop/.circleci/config.yml#L163

Additionally, in the OWASP documentation they state for the Baseline Scan:
> This script is intended to be ideal to run in a CI/CD environment

The biggest difference is that the baseline script crawls through the target to find any vulnerabilities but does not actually execute them - while the full scan does actually attempt to execute the attacks.

While it is valuable to have a full scan completed periodically it doesn't need to be done with *every* CI check and not necessarily against Docker builds that won't be used in Cloud.gov once the buildpacks work is merged in.

Instead, I'd suggest that we change the CI OWASP step to use the Baseline Scan option for all branches except for `raft-tdp-main`. When a PR is opened from `raft-tdp-main` it would be targeting the HHS repo per our gitflow, meaning that the work is approved and has been deployed to the staging environment (Once #847 is completed). At this point we could use a different step in Circle CI that would instead run a ZAP Full Scan and point it directly towards the live staging site. This gets us automated pen-testing against a production-like instance and should lower the time it takes to run through CI steps for in progress PRs, which should help prevent these intermittent timeout issues.

**Comment by jtwillis92** [comment]
@kniz-raft @ADPennington we should discuss this one in backlog refinement - I think we could get a lot more reliability in CI/CD if we move to the baseline scan.

**Comment by kniz-raft** [comment]
Need a decision on a direction for this.

**Comment by jtwillis92** [comment]
Closing in favor of #1032
